python/smqtk/algorithms/saliency/sbsm.py

```diff
@@ -0,0 +1,208 @@
+import numpy as np
+from tqdm import tqdm
+import copy
+from PIL import Image
+import os
+from sklearn.metrics.pairwise import euclidean_distances
+import six
+from smqtk.algorithms import SmqtkAlgorithm
+from smqtk.algorithms.saliency import SaliencyBlackbox,ImageSaliencyAugmenter
+from smqtk.representation.data_element.memory_element import DataMemoryElement
+
+class SBSM_SaliencyBlackbox (SaliencyBlackbox):
+    
+    """
+    Blackbox function that produces some floating point scalar value for a
+    given descriptor element.
+    :param query_f: Feature of query image
+    :param type: smqtk.representation.DescriptorElement
+    :param base_descr: Base image descriptor
+    :param type: smqtk.representation.DescriptorElement
+        """
+    def __init__(self, query_f, base_descr):
+        
+        self.query_f = query_f
+        
+        self.base_descr = base_descr
+
+    @classmethod
+    def is_usable(cls):
+        """
+        Check whether this implementation is available for use.
+        Required valid presence of query image feature and base image descriptor
+        :return:
+            Boolean determination of whether this implementation is usable.
+        :rtype: bool
+        """
+        
+        return self.base_descr and self.query_f
```

Comment:
This is not valid as ``self`` is not defined here. The intention of this class method is more of a systematic "is usable". Since this implementation does not use anything other than dependencies stated in SMQTK's base requirements, this method can just return ``True``.

Reply:
Yes, it was also showed up as a run time error, Changed it.